examples/generate.sh

```diff
@@ -94,7 +93,10 @@ if [[ -n "${3-}" ]] && [[ $3 != -* ]] && [[ $3 != --* ]]; then
     exit 1
   fi
 else
-  echo "Generating yaml for \"single-node\" cluster by default"
```

Comment:
@jichenjc @sbueringer I had earlier set `single-node` as default but would leed to weird implementation. 

If we set "single-node" as default, it would mean that command like `./generate.sh clouds.yaml openstack <folder directory>` would be invalid. 
Default "single-node" would only work in case we dont give out folder directory Eg. "./generate.sh clouds.yaml openstack"

We can have 2 ways out:
1. Don't have a default value for control-plane preference and ask user to mandatory ask for preference. (update docs for that)
2. Switch order of user input. Eg. If we can take something like `./generate.yaml <clouds.yaml> <cloud> <out directory> <controlplane preference>`. (which would also leed to the same situation for out directory)

Reply:
The problem is it's not straightforward to support multiple optional parameters this way. It would be possible to call it like this `./generate.yaml <clouds.yaml> <cloud> "" <out directory>` Then the default value would be used. But that's also not really convenient.

But for me it would be okay to change the order and declare the output folder as mandatory parameter. In a lot of other scripts/tools it is. It's not really intuitive that _out is a default :)

Reply:
sounds good !!